pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1548,19 +1548,29 @@ func useCSIPlugin(vpm *volume.VolumePluginMgr, spec *volume.Spec) bool {
 }
 
 func translateSpec(spec *volume.Spec) (*volume.Spec, error) {
+	if spec.PersistentVolume == nil && spec.Volume == nil {
+		return &volume.Spec{}, fmt.Errorf("not a valid volume spec")
+	}
+	var csiPV *v1.PersistentVolume
+	var err error
+	inlineVolume := false
 	if spec.PersistentVolume != nil {
 		// TranslateInTreePVToCSI will create a new PV
-		csiPV, err := csilib.TranslateInTreePVToCSI(spec.PersistentVolume)
+		csiPV, err = csilib.TranslateInTreePVToCSI(spec.PersistentVolume)
 		if err != nil {
 			return nil, fmt.Errorf("failed to translate in tree pv to CSI: %v", err)
 		}
-		return &volume.Spec{
-			PersistentVolume: csiPV,
-			ReadOnly:         spec.ReadOnly,
-		}, nil
 	} else if spec.Volume != nil {
-		return &volume.Spec{}, fmt.Errorf("translation is not supported for in-line volumes yet")
-	} else {
-		return &volume.Spec{}, fmt.Errorf("not a valid volume spec")
+		// TranslateInTreeVolumeToCSI will create a new PV
+		csiPV, err = csilib.TranslateInTreeVolumeToCSI(spec.Volume)
```

Comment:
many instances of `InTreeVolume` should be changed to `InlineVolume` I think

Reply:
So I was trying to distinguish `TranslateInTreeVolumeToCSI` from `TranslateInTreePVToCSI`  as the former handles `spec.Volume` (associated with inline) while the latter handles `Spec.PersistentVolume` (associated with PV). I understand the fact that it may not be clear that `TranslateInTreeVolumeToCSI` is meant to handle inline.

Maybe we can be a bit verbose and call the in-tree in-line translator: `TranslateInTreeInlineVolumeToCSI`?

Reply:
oh I see now, yea lets err on the side of verbosity. It's really long but also this feature is inherently confusing so I think its better we fully qualify names so its marginally easier to understand